Add aperiodic monotiles to landing page

Splines opened the pull request:
A fresh and interactive landing page look for the next releases (starting with 1.9.0 ~~#546~~ #561 ) using the newly found aperiodic monotiles, see the [official web page](https://cs.uwaterloo.ca/~csk/hat/) and the [repo with the BSD3-licensed code](https://github.com/isohedral/hatviz).

![image](https://github.com/MaMpf-HD/mampf/assets/37160523/a191cad9-4c01-495d-b769-e14f10e7e806)


Different colors (which I prefer as the overall tone is more balanced):

![image](https://github.com/MaMpf-HD/mampf/assets/37160523/5ee506ba-1945-4bbd-bb6e-ec81c2080366)



## Issues

- Matrix multiplication is implemented in JS, that's why it is a bit slow. Try to hit Reset, then 2x "Monotile". The pattern will then not fill up the whole screen but is far more responsive. We could also have it initialized this way...
![image](https://github.com/MaMpf-HD/mampf/assets/37160523/7eaa24d6-be24-4daf-b441-4473559b43e8)


- On tablets, we cannot scroll the website anymore as the dragging of the canvas is somehow taking precedence.
- I've changed one color to red. If you don't like it, we can change it back to grey as it is [here](https://cs.uwaterloo.ca/~csk/hat/app.html).

Splines commented:
@fosterfarrell9 I've added the aperiodic monotiles (ein stein) as interactive background. Currently available on [experimental](https://mampf-experimental.mathi.uni-heidelberg.de/) for you to try out. More infos in the PR.

fosterfarrell9 commented:
Is it possible to make the tiles bigger? I think that the current size makes it look a little restless/"unruhig".

Splines commented:
@fosterfarrell9 I've made the tiles bigger and darkened the colors to make the background less busy/jarring. Due to the bigger tiles, the numer of tiles generated are reduced which also makes the page load faster.
I've also fixed the issue on mobile devices, it should work on all devices now (buttons are disabled on small devices and you cannot drag the background, otherwise one could not scroll the site anymore). I've also implemented a small logic to disable the "monotile" button as to not freeze browsers when generating too many tiles.

![image](https://github.com/MaMpf-HD/mampf/assets/37160523/ba15fb86-502c-4416-8560-a1b8f8e869a7)

![image](https://github.com/MaMpf-HD/mampf/assets/37160523/e8e14d1e-d5e4-4e91-9116-8054f5c7c2bc)

fosterfarrell9 commented on app/assets/javascripts/monotile/hat.js:
```diff
@@ -0,0 +1,466 @@
+	'T': [251, 251, 251],
+	'P': [250, 250, 250],
+	'F': [191, 191, 191]
+};
```
I have shown the new landing screen to several people. There were some complaints about the dark grey. I experimented myself a little with the colors and arrived at these values:
```javascript
const colors = {
	'H1': [0, 137, 212],
	'H': [148, 205, 235],
	'T': [251, 251, 251],
	'P': [229, 234, 241],
	'F': [230, 255, 255]
};
```
Are they better? I do not know, we would have to ask some more people.

Splines replied:
I don't like the 'F' color where you just lowered the red value (starting off of white), which results in a blue-greenish look that is too greenish in my eyes.
![image](https://github.com/MaMpf-HD/mampf/assets/37160523/829a731e-3463-4710-b6ac-d75f0a264f3c)

If the argument was that the grey is too prominent, we could just replace it with a more light grey, so that the blue tiles attract more attention:

```js
const colors = {
	'H1': [0, 137, 212],
	'H': [148, 205, 235],
	'T': [251, 251, 251],
	'P': [250, 250, 250],
	'F': [220, 220, 220]
};
```

It looks like this which I think is a workable solution:

![image](https://github.com/MaMpf-HD/mampf/assets/37160523/d442ea52-d85d-4393-8e22-4cf91e9098ea)


Other than that, I thought one opinion is that the background shouldn't be too bright and not too colorful because MaMpf "is sober and should represent clarity and focus" [you or Hendrik Kasten]?. That's why I switched to the grey in the first place to have more subdued colors (even though I'd love to see the red color, but I agree that it is too much).

What were the complaints about the grey? This color is even is built-into the Mathematikon as facing concrete everywhere, so it perfectly represents the mathematical faculty ;) Of course, color choice is very subjective in the end.

fosterfarrell9 replied:
I like the colors of the previous comment, will approve if you use them.

Splines commented:
@fosterfarrell9 Requesting a review just for the last two commits where I merged the `lint/eslint-all` branch into this branch. You probably want to review #575 first (e.g. maybe some rules will change due to your review and then we'd have to update here accordingly).
